### PR TITLE
[Bug fix] Fix reversed `assert_with_ulp` arguments in TTNN tests

### LINKED ISSUE #55657: Reversed `assert_with_ulp` arguments in TTNN tests
# Summary

Several tests pass device output first and the reference second to
`assert_with_ulp(expected_result, actual_result, ...)`.
This comparison is asymmetric: it measures `abs(actual - expected) / ULP(expected)`
using the actual tensor's dtype for the ULP spacing. Reversing the arguments can
therefore hide accuracy regressions or cause false failures, particularly across
power-of-two boundaries. This is a call-site bug, not a request to change the metric.

## Minimal reproducer

Run in the repository's Python environment; no accelerator is needed:

```python
import torch
from tests.ttnn.utils_for_testing import assert_with_ulp

expected = torch.tensor([0.9999998807907104], dtype=torch.float32)
actual = torch.tensor([1.0], dtype=torch.float32)

assert_with_ulp(actual, expected, ulp_threshold=1)  # Incorrect order: passes (1 ULP).
assert_with_ulp(expected, actual, ulp_threshold=1)  # Correct order: fails (2 ULP).
```

This behavior was reproduced on the host; the analogous boundary case also
reproduces with BF16, FP16, and FP64.

## Affected tests

A repository-wide audit of 257 calls in 64 files found 51 reversed call sites.
All 51 also exist on `origin/main` at `e90ebd676b405a026ea30c5153b8bf79a99d9001`.

Paths below are relative to `tests/ttnn/`:

| File | Reversed calls |
| --- | ---: |
| `unit_tests/operations/eltwise/test_unary.py` | 16 |
| `unit_tests/operations/eltwise/test_unary_ops_ttnn.py` | 12 |
| `unit_tests/operations/eltwise/test_binary_int32.py` | 10 |
| `unit_tests/operations/eltwise/test_binary_composite.py` | 3 |
| `unit_tests/operations/eltwise/test_binaryng_ND.py` | 2 |
| `unit_tests/operations/eltwise/test_div_ops.py` | 2 |
| `unit_tests/operations/eltwise/test_unary_sharding.py` | 2 |
| `unit_tests/operations/eltwise/test_ternary.py` | 2 |
| `nightly/unit_tests/operations/eltwise/test_unary_shard.py` | 1 |
| `nightly/unit_tests/operations/eltwise/test_ternary_sharded.py` | 1 |

Of these, 41 are reachable, positive-tolerance comparisons. The other ten are
five dormant dtype branches, two zero-tolerance checks, and three checks followed
by exact equality assertions. Not every reversed call currently changes pass/fail.

## Proposed fix

- Put the reference first at the confirmed sites, preferably using explicit
  `expected_result=` and `actual_result=` keywords.
- Add a host-only regression covering the asymmetric power-of-two boundary.
- Rerun the affected device tests with their existing thresholds. Investigate
  any newly exposed failures rather than automatically relaxing tolerances.

Device-test impact has not yet been validated for the proposed swaps.

### PR DESCRIPTION
**Pull request by jasondavies**
### PR Category

Bug fix

### Summary

Reversing the arguments for `assert_with_ulp` can hide accuracy regressions or cause false failures, particularly across power-of-two boundaries.

Replace with explicit keyword arguments everywhere.

Closes https://github.com/tenstorrent/tt-metal/issues/55657

### Notes for reviewers

Waiting for https://github.com/tenstorrent/tt-metal/pull/55506 to merge.

### CODE REVIEW

**Review by mateusznowakTT — APPROVED** [review]

**Comment by mateusznowakTT** [comment]
@jasondavies : Thanks for the PR. Since #55506 you marked as dependency is merged, I'm taking this out of draft - I hope this is fine for you.

====
Also, few cases failed in Sanity tests, if you could please fill the gaps, it would be great. Thanks !

**Review by mateusznowakTT — APPROVED** [review]

**Comment by jasondavies** [comment]
@mateusznowakTT Done

**Review by nmauriceTT — APPROVED** [review]

**Comment by jasondavies** [comment]
@mateusznowakTT Please ping codeowners or get the necessary bypass approvals -- unfortunately there's quite a few to ping.  If it's too awkward to merge I guess we can drop the TypeError temporarily.

Edit: I guess it's not that many codeowners, I've seen worse :)

**Review by mateusznowakTT — APPROVED** [review]

**Review by s-jovic — APPROVED** [review]

**Review by Aswinmcw — APPROVED** [review]

**Review by KalaivaniMCW — APPROVED** [review]
Thanks for this change, keyword args is better.  Please let me know if you see any failures on Eltwise or TM ops tests that needs debugging. 
I'll post the nightly L2 CI link below 
Nightly L2 for eltwise and data movement: https://github.com/tenstorrent/tt-metal/actions/runs/34211099370

**Review by bbradelTT — APPROVED** [review]

**Review by itarabanTT — APPROVED** [review]

**Comment by jasondavies** [comment]
@Aswinmcw @KalaivaniMCW Ready for merge

**Comment by mateusznowakTT** [comment]
@jasondavies : There are few new cases that require alignment (see failed merge gate).
Will this be fine with you if I contribute to this PR to make sure it goes through pipelines without delays, or you'd rather do all the adjustments yourself ?

**Comment by jasondavies** [comment]
@mateusznowakTT Fixed new `assert_with_ulp` calls that were added to main.  Feel free to merge again

**Review by TT-BrianLiu — APPROVED** [review]

**Comment by jasondavies** [comment]
@mateusznowakTT Retry merge I guess?  Looks like CI is not behaving
